### PR TITLE
Fix change show location not creating show folder

### PR DESCRIPTION
**Pull request by fernandog**
Fix change show location not creating show folder when setting 'CREATE_MISSING_SHOW_DIRS' is set

### CODE REVIEW

File: medusa/server/web/home/handler.py [hunk 1]
```diff
@@ -1577,26 +1577,28 @@ def editShow(self, show=None, location=None, allowed_qualities=None, preferred_q
...
+                        os.mkdir(new_location)
+                        helpers.chmod_as_parent(show_obj._location)
                         show_obj.location = location
+                    except (OSError, IOError) as e:
+                        log_message = u"Unable to create the show directory '{location}. " \
+                                      u"Error: {error}".format(location=new_location, error=ex(e))
+                        logger.log(log_message, logger.DEBUG)
+                        errors.append(log_message)
+
+                    if do_update and os.path.isdir(show_obj._location):
```
Comment (labrys) [inline medusa/server/web/home/handler.py]: Your reverasal of the condition logic is now requiring two checks of `isdir(show_obj._location)`

File: medusa/server/web/home/handler.py [hunk 2]
```diff
@@ -1577,26 +1577,28 @@ def editShow(self, show=None, location=None, allowed_qualities=None, preferred_q
...
-                elif not do_update:
-                    # change it
+                logger.log('Changing show location to: {new}'.format(new=new_location), logger.DEBUG)
+                if not os.path.isdir(show_obj._location) and app.CREATE_MISSING_SHOW_DIRS:
+                    logger.log(u"Show directory doesn't exist, creating it", logger.DEBUG)
                     try:
+                        os.mkdir(new_location)
+                        helpers.chmod_as_parent(show_obj._location)
                         show_obj.location = location
+                    except (OSError, IOError) as e:
```
Comment (labrys) [inline medusa/server/web/home/handler.py]: You're  doing too much inside the `try`.  Try should wrap the minimum code that could raise the exception.

**Comment by labrys** [comment]
@fernandog does this address a specific issue?  if so could you please reference it.  if not could you please explain the issue and steps required to reproduce.

**Comment by fernandog** [comment]
@labrys as in PR body: show location not creating show folder when setting 'CREATE_MISSING_SHOW_DIRS' is set.

Reproduce: mass update shows and change root to a destination where there is no show folders. Try it.

File: medusa/server/web/home/handler.py (hunk 1 above)
Comment (fernandog) [inline medusa/server/web/home/handler.py]: @labrys yes, it show was created, we need to check again. What if it couldn't create for some error?
we can't call a show refresh

File: medusa/server/web/home/handler.py (hunk 2 above)
Comment (fernandog) [inline medusa/server/web/home/handler.py]: @labrys too much!? I'm creating dir and doing chmod

File: medusa/server/web/home/handler.py (hunk 1 above)
Comment (labrys) [inline medusa/server/web/home/handler.py]: makes sense

File: medusa/server/web/home/handler.py (hunk 2 above)
Comment (labrys) [inline medusa/server/web/home/handler.py]: `os.mkdir` shouldn't raise `IOError` and `helpers.chmod_as_parent` already catches its own `OSError` so it should be 

```python
                    try:
                        os.mkdir(new_location)
                    except OSError as e:
                        log_message = u"Unable to create the show directory '{location}. " \
                                      u"Error: {error}".format(location=new_location, error=ex(e))
                        logger.log(log_message, logger.DEBUG)
                        errors.append(log_message)
                    else:
                        helpers.chmod_as_parent(show_obj._location)
                        show_obj.location = location
```

**Comment by labrys** [comment]
I'll try to make time to test it today, but I may be out of the house all day.

**Comment by fernandog** [comment]
changes implemented

**Comment by fernandog** [comment]
```
2017-03-23 12:23:59 INFO     Thread-24 :: [b4a8a54] For show Vikings changing dir from /media/SAMSUNG/media/series/Vikings to C:\Users\fernando\Documents\Series\Vikings
2017-03-23 12:23:59 DEBUG    Thread-24 :: [b4a8a54] Changing show location to: C:\Users\fernando\Documents\Series\Vikings
2017-03-23 12:23:59 DEBUG    Thread-24 :: [b4a8a54] Show directory doesn't exist, creating it
2017-03-23 12:23:59 DEBUG    Thread-24 :: [b4a8a54] TVDBv2 260449: Setting location: C:\Users\fernando\Documents\Series\Vikings
2017-03-23 12:23:59 DEBUG    Thread-24 :: [b4a8a54] 260449: Saving to database: Vikings
```

**Comment by fernandog** [comment]
Fixes: https://github.com/pymedusa/Medusa/issues/2079

**Pull request merged**